src/engine/source/base/include/base/utils/threadEventDispatcher.hpp

```diff
@@ -14,14 +14,28 @@
 
 #include "rocksDBQueue.hpp"
 #include "rocksDBQueueCF.hpp"
-#include "threadSafeMultiQueue.hpp"
 #include "threadSafeQueue.hpp"
 #include <atomic>
 #include <iostream>
 #include <thread>
 
 constexpr auto UNLIMITED_QUEUE_SIZE = 0;
+constexpr uint8_t SINGLE_THREAD = 1;
+constexpr uint8_t MULTI_THREAD = 3;
 
+enum class ThreadEventDispatcherType
+{
+    SINGLE_THREADED_ORDERED,
+    MULTI_THREADED_UNORDERED
+};
+
+/**
+ * @brief Class in charge to dispatch events in a queue to be processed by a functor.
+ *
+ * The Functor function implemented by a caller method will be in charge to process the events
```

Comment:
```suggestion
 * @note The Functor function implemented by a caller method will be in charge of processing the events. In case of an
 * exception, all events that remain will be re-inserted into the queue, therefore, the Functor is responsible for
 * handling the exceptions and dropping the events if necessary.
```